extensions/VisualEditor/lib/ve/lib/jquery.client/jquery.client.js

```diff
@@ -0,0 +1,299 @@
+
+				// Name of browsers or layout engines we don't recognize
+				uk = 'unknown',
+				// Generic version digit
+				x = 'x',
+				// Strings found in user agent strings that need to be conformed
+				wildUserAgents = ['Opera', 'Navigator', 'Minefield', 'KHTML', 'Chrome', 'PLAYSTATION 3', 'Iceweasel'],
+				// Translations for conforming user agent strings
+				userAgentTranslations = [
+					// Tons of browsers lie about being something they are not
+					[/(Firefox|MSIE|KHTML,?\slike\sGecko|Konqueror)/, ''],
+					// Chrome lives in the shadow of Safari still
+					['Chrome Safari', 'Chrome'],
+					// KHTML is the layout engine not the browser - LIES!
+					['KHTML', 'Konqueror'],
+					// Firefox nightly builds
+					['Minefield', 'Firefox'],
+					// This helps keep different versions consistent
+					['Navigator', 'Netscape'],
+					// This prevents version extraction issues, otherwise translation would happen later
+					['PLAYSTATION 3', 'PS3']
+				],
+				// Strings which precede a version number in a user agent string - combined and used as
+				// match 1 in version detection
+				versionPrefixes = [
+					'camino', 'chrome', 'firefox', 'iceweasel', 'netscape', 'netscape6', 'opera', 'version', 'konqueror',
+					'lynx', 'msie', 'safari', 'ps3', 'android'
+				],
+				// Used as matches 2, 3 and 4 in version extraction - 3 is used as actual version number
+				versionSuffix = '(\\/|\\;?\\s|)([a-z0-9\\.\\+]*?)(\\;|dev|rel|\\)|\\s|$)',
+				// Names of known browsers
+				names = [
+					'camino', 'chrome', 'firefox', 'iceweasel', 'netscape', 'konqueror', 'lynx', 'msie', 'opera',
+					'safari', 'ipod', 'iphone', 'blackberry', 'ps3', 'rekonq', 'android'
+				],
+				// Tanslations for conforming browser names
+				nameTranslations = [],
+				// Names of known layout engines
+				layouts = ['gecko', 'konqueror', 'msie', 'trident', 'opera', 'webkit'],
+				// Translations for conforming layout names
+				layoutTranslations = [ ['konqueror', 'khtml'], ['msie', 'trident'], ['opera', 'presto'] ],
+				// Names of supported layout engines for version number
+				layoutVersions = ['applewebkit', 'gecko', 'trident'],
+				// Names of known operating systems
+				platforms = ['win', 'wow64', 'mac', 'linux', 'sunos', 'solaris', 'iphone'],
+				// Translations for conforming operating system names
+				platformTranslations = [ ['sunos', 'solaris'], ['wow64', 'win'] ],
+
+				/* Methods */
+
+				/**
+				 * Performs multiple replacements on a string
+				 */
+				translate = function ( source, translations ) {
+					var i;
+					for ( i = 0; i < translations.length; i++ ) {
+						source = source.replace( translations[i][0], translations[i][1] );
+					}
+					return source;
+				},
+
+				/* Pre-processing */
+
+				ua = nav.userAgent,
+				match,
+				name = uk,
+				layout = uk,
+				layoutversion = uk,
+				platform = uk,
+				version = x;
+
+			if ( match = new RegExp( '(' + wildUserAgents.join( '|' ) + ')' ).exec( ua ) ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "match" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)